app/src/components/domain/RiskImminentEventMap/index.tsx

```diff
@@ -248,6 +322,53 @@ function RiskImminentEventMap<
         [allIconsLoaded],
     );
 
+    const activeLayerOptions = useMemo(() => {
+        if (isNotDefined(activeLayersMapping)) {
+            return [];
+        }
+        return layerOptions.filter((opt) => activeLayersMapping[opt?.key]);
+    }, [activeLayersMapping, layerOptions]);
+
+    const popupDetails = useMemo(() => {
+        const eventDetails = clickedPointProperties
+            ? clickedPointProperties.feature.properties
+            : undefined;
+        return eventDetails;
+    }, [clickedPointProperties]);
+
+    const visibleAllHazardPoints = activeEvent?.hazard_type !== 'TC' || isNotDefined(activeEventFootprint);
+
+    const mapOrder = useMemo(() => {
+        const hasLayerValue = Object.values(layers)?.some(isDefined);
```

Comment:
Do we need to check for hasLayerValue?

Reply:
Also why not just mount 2 different MapOrder on the render itself.

Reply:
This is intended to hide all the hazard point when a particular cyclone is clicked for more detail because when multiples cyclone occur side by side, the visualization become confusing.